src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,113 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl service;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        service = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setId(474797L);
+        user.setLogin("userLogin47");
+        user.setPassword("password1111");
+        user.setAge(25);
+    }
+
+    @Test
+    void register_UserValid_Ok() {
+        StorageDao storageDao = new StorageDaoImpl();
+        assertNotNull(service.register(user));
+        assertNotNull(storageDao.get(user.getLogin()));
+    }
+
+    @Test
+    void register_UserNullValue_NotOk() {
+        service.register(null);
+        assertThrows(InvalidUserException.class, () -> service.register(null),
+                "User must not be null");
+    }
+
+    @Test
+    void register_LoginNotAvailable_NotOk() {
+        StorageDao storageDao = new StorageDaoImpl();
+        storageDao.add(user);
+        assertNotNull(storageDao.get(user.getLogin()));
+        assertThrows(InvalidUserException.class, () -> service.register(user),
+                "A user with this login is already registered");
```

Comment:
You should not use `service.register(user)` to add a user to the storage for testing purposes. Instead, use `storageDao.add(user)` directly to prepare your test environment, as the `register` method is the one under test.